### PR TITLE
Fix Deprecation

### PR DESCRIPTION
**Pull request by mikehardy**
Make use of deprecated APIs a build error, and fix them all now or at least log them as side-issues

This is still building in CI and needs a lot more testing before I merge it, but I wanted to make this public since it's ginormous.

After a couple days playing with this locally (in daily use for me) and also re-reading important parts of the diff I'll ask for this to merge for real

### CODE REVIEW

**Comment by timrae** [comment]
Just merge when you're happy with it, if there are no changes to the business logic then I'm unlikely to review it.

**Comment by mikehardy** [comment]
That almost makes me more nervous, but that's probably a good thing. No intention to change logic but I'll obviously fix if something goes wrong. I'll work through the diffs off an on a couple more times then merge it in then. For codacy: It'll be a net-negative codacy change but the way codacy ascribes ownership means the PR will be dinged for something I don't want to take the time out to fix this round. Worth the progress though I think

**Comment by timrae** [comment]
Feel free to ping me if you need a second opinion on any specific points

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 1]
```diff
@@ -229,7 +227,7 @@ public void onSelection(String searchName) {
                 mSearchTerms = savedFiltersObj.optString(searchName);
                 Timber.d("OnSelection using search terms: %s", mSearchTerms);
                 mSearchView.setQuery(mSearchTerms, false);
-                MenuItemCompat.expandActionView(mSearchItem);
+                mSearchItem.expandActionView();
                 searchCards();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java]: Not sure if this is the cause - but something isn't right in CardBrowser and I think it's in this diff. Reproduction: multi-select works initially (all decks) but if you select a deck then long-press on a card the loading animation shows up and checkbox selection is totally broken.

File: api/build.gradle [hunk 2]
```diff
@@ -43,6 +43,15 @@ task androidJavadocs(type: Javadoc) {
...
 
+// Deprecation is an error. Use @SuppressWarnings("deprecation") and justify in the PR if you must
+tasks.withType(JavaCompile) {
+    options.compilerArgs << "-Xlint:deprecation" << "-Xmaxwarns" << "1000" << "-Werror"
+}
+
+tasks.withType(JavaCompile) {
+    options.compilerArgs << "-Xlint:deprecation" << "-Xmaxwarns" << "1000"
+}
+
```
Comment (mikehardy) [inline api/build.gradle]: obviously duplication error - fixed

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java]: It's not this diff. Same thing happens on the emulator I saw it on with CardBrowser using hotfix-2.8.5 branch. Fresh emulator same API (28) works fine with this branch and hotfix-2.8.5. I'll save that emulator for examination but release this PR

**Comment by mikehardy** [comment]
CardBrowser issue was an emulator issue. Fresh emulator same API28 was fine, other emulator APIs and device fine. No reason I can see to wait

**Comment by mikehardy** [comment]
All the things that are touchy are pushed into separate issues, I don't think re-reading it a 3rd time will help so I held my breath and hit rebase-and-merge on this one.

**Pull request merged**